### PR TITLE
feat: add error-boundary & error-indicator components

### PR DESCRIPTION
**Pull request by Aliaksandr-Shliayeu**
### Error handler for "hard errors"

##Includes:

-                 NEW <ErrorBoundy /> component
-                NEW < ErrorIndicator /> component
-                UPDATED < App /> component

image

### CODE REVIEW

File: js_machine_front/src/components/PartnersLogo.tsx [hunk 1]
```diff
@@ -3,10 +3,10 @@ import 'styles/partnersLogo.css';
 
 export const PartnersLogo: React.FC = memo(() => {
     return (
-        <div className="partners-logo">
```
[the remark this answers is not in the document]
Comment (Aliaksandr-Shliayeu) [reply js_machine_front/src/components/PartnersLogo.tsx]: Probably not, but following the "react" best practice, empty tags should be self-closing. Also, I have deleted unnecessary spaces at the end of the string.

File: js_machine_front/src/components/error-boundry/error-boundry.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+
+import ErrorIndicator from '../error-indicator/error-indicator';
+
+interface ErrorBoundryState {
+  hasError: boolean;
+}
+
+export default class ErrorBoundry extends React.Component<{}, ErrorBoundryState> {
```
[the remark this answers is not in the document]
Comment (Aliaksandr-Shliayeu) [reply js_machine_front/src/components/error-boundry/error-boundry.tsx]: Removed defaults exports from ErrorBoundry and ErrorIndicator components

File: js_machine_front/src/components/error-indicator/index.ts [hunk 3]
```diff
@@ -1,3 +1 @@
-import ErrorIndicator from './error-indicator';
-
-export {ErrorIndicator};
+export { ErrorIndicator } from './error-indicator';
```
[the remark this answers is not in the document]
Comment (Aliaksandr-Shliayeu) [reply js_machine_front/src/components/error-indicator/index.ts]: Fixed

File: js_machine_front/src/components/error-boundry/index.ts [hunk 4]
```diff
@@ -1,3 +1 @@
-import ErrorBoundry from './error-boundry';
-
-export {ErrorBoundry};
+export {ErrorBoundry} from './error-boundry';
```
[the remark this answers is not in the document]
Comment (Aliaksandr-Shliayeu) [reply js_machine_front/src/components/error-boundry/index.ts]: Fixed

File: js_machine_front/src/components/error-boundry/error-boundry.tsx [hunk 5]
```diff
@@ -25,3 +24,5 @@ export default class ErrorBoundry extends React.Component<{}, ErrorBoundryState>
     return this.props.children;
   }
 }
+
+export { ErrorBoundry };
```
[the remark this answers is not in the document]
Comment (Aliaksandr-Shliayeu) [reply js_machine_front/src/components/error-boundry/error-boundry.tsx]: Fixed

File: js_machine_front/src/components/error-boundry/error-boundry.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
```
Comment (vharadkou) [inline js_machine_front/src/components/error-boundry/error-boundry.tsx]: typo, rename component to error-boundary

**Review by vharadkou — CHANGES_REQUESTED** [review]

File: js_machine_front/src/components/error-boundry/error-boundry.tsx (hunk 6 above)
Comment (Aliaksandr-Shliayeu) [reply js_machine_front/src/components/error-boundry/error-boundry.tsx]: Fixed

**Review by vharadkou — APPROVED** [review]

**Pull request merged**